src/gossip/service.zig

```diff
@@ -1585,33 +1598,38 @@ pub const GossipService = struct {
             pubkey_to_endpoint.deinit();
         }
 
+        // pre-allocate memory to track insertion failures
+        var max_inserts_per_push: usize = 0;
+        for (batch_push_messages.items) |*push_message| {
+            max_inserts_per_push = @max(max_inserts_per_push, push_message.gossip_values.len);
+        }
+        var failed_insert_indexs = try std.ArrayList(usize)
+            .initCapacity(self.allocator, max_inserts_per_push);
+        defer failed_insert_indexs.deinit();
+
         // insert values and track the failed origins per pubkey
         {
             var gossip_table_lock = self.gossip_table_rw.write();
             defer gossip_table_lock.unlock();
-
             for (batch_push_messages.items) |*push_message| {
                 var gossip_table: *GossipTable = gossip_table_lock.mut();
                 const valid_len = self.filterBasedOnShredVersion(
                     gossip_table,
                     push_message.gossip_values,
                     push_message.from_pubkey.*,
                 );
-
-                var result = try gossip_table.insertValues(
+                try gossip_table.insertValuesMinAllocs(
                     push_message.gossip_values[0..valid_len],
                     GOSSIP_PUSH_MSG_TIMEOUT_MS,
-                    false,
-                    false,
+                    &failed_insert_indexs,
                 );
-                const failed_insert_indexs = result.failed.?;
-                defer failed_insert_indexs.deinit();
 
-                self.logger
-                    .field("n_values", valid_len)
-                    .field("from_addr", &push_message.from_pubkey.string())
-                    .field("n_failed_inserts", failed_insert_indexs.items.len)
-                    .info("gossip: recv push_message");
+                // logging this message takes too long and causes a bottleneck
```

Comment:
instead of commenting this out - maybe we create a new logging level which is not the default? - likely because the other recv logs will also be slowing it down

might also be ok to leave it commented out for now but thinking more long-term - is the solution:
- fix the logging bottleneck to support this large freq of messages - (this is probs the long term soln)
- dont log too much (hard to say how much value these bring tbh) - (probs keep this in the back of our minds now that we know its a bottleneck - _70%!_)
- modify the log level to only print these if the user wants - (probs a good quick fix)

Reply:
All three sound like good ideas.

> modify the log level to only print these if the user wants - (probs a good quick fix)

70% of the time spent logging is what I saw when I was already filtering out the messages by setting the log level to warn.

Calling `field` results in an allocation regardless of the log level. We may want to change the logging so you specify the level somehow before calling field. Also it could be useful to introduce a compile-time max log level.

Reply:
I'm running the benchmark some more and seeing some counter-intuitive behavior.

With debug logs **disabled**, the timer still reports about 70% of the time is used during the single log statement.

With debug logs **enabled**, I would expect the logger to consume even more time, but it's actually reporting a bit less, more like 50-60% of the time the function is running.